src/auth.py

```diff
@@ -25,7 +23,7 @@ def initialize_token():
 
 def authenticate(func):
     def wrapper(*args, **kwargs):
-        self = args[0]  # type: src.handlers.BaseAPIHandler
+        self = args[0]
```

Comment:
Please please don't assign `self`. It breaks all sorts of things.

Create an abstraction around it or just another name.

Reply:
Isn't `self` just a variable here since this function is not an instance method? Github highlights it as a keyword but assigning `self` here does not break anything as such as far as I know. It is a mere variable with limited scope. This wrapper function returns `func` and the variable self would not interplay in any way with that function. Pycharm too does not complain about doing this (it always complains when we assign keywords or any such global function which can lead to unexpected behavior).

Reply:
Hmm, okay. Decorators usually make sense out of context too. If you are meaning that this decorator should be used in a place where args[0] is `self` then you should note it.

It's confusing for a first time reader, because the decorators themselves are tightly coupled with their usage.

Reply:
Makes sense. Changed the variable name for more clarity. Yes, these decorators are very coupled with the API and need the API Handler (args[0]) to do their job. Should be fixed.